src/openloco/windows/construction/OverheadTab.cpp

```diff
@@ -0,0 +1,387 @@
+    static void on_dropdown(window* self, widget_index widgetIndex, int16_t itemIndex)
+    {
+        if (widgetIndex != widx::track_dropdown)
+            return;
+
+        if (itemIndex != -1)
+        {
+            _lastSelectedTrackModSection = itemIndex;
+            self->invalidate();
+        }
+    }
+
+    // 0x0049ECD1
+    static void on_update(window* self)
+    {
+        common::on_update(self, (1 << 5));
+    }
+
+    // 0x0049EC15
+    static void on_tool_update(window& self, const widget_index widgetIndex, const int16_t x, const int16_t y)
+    {
+        registers regs;
+        regs.esi = (int32_t)&self;
+        regs.dx = widgetIndex;
+        regs.ax = x;
+        regs.bx = y;
+        call(0x0049EC15, regs);
+    }
+
+    // 0x0049EC20
+    static void on_tool_down(window& self, const widget_index widgetIndex, const int16_t x, const int16_t y)
+    {
+        registers regs;
+        regs.esi = (int32_t)&self;
+        regs.dx = widgetIndex;
+        regs.ax = x;
+        regs.bx = y;
+        call(0x0049EC20, regs);
+    }
+
+    // 0x0049E7D3
+    static void prepare_draw(window* self)
+    {
+        common::prepare_draw(self);
+
+        auto activatedWidgets = self->activated_widgets;
+        activatedWidgets &= ~(1 << widx::checkbox_1 | 1 << widx::checkbox_2 | 1 << widx::checkbox_3 | 1 << widx::checkbox_4);
+
+        self->widgets[widx::checkbox_1].type = widget_type::none;
+        self->widgets[widx::checkbox_2].type = widget_type::none;
+        self->widgets[widx::checkbox_3].type = widget_type::none;
+        self->widgets[widx::checkbox_4].type = widget_type::none;
+
+        if (_trackType & (1 << 7))
+        {
+            auto trackType = _trackType & ~(1 << 7);
+            auto roadObj = objectmgr::get<road_object>(trackType);
+
+            auto args = FormatArguments();
+            args.push(roadObj->name);
+
+            if (_modList[0] != 0xFF)
+            {
+                auto roadExtraObject = objectmgr::get<road_extra_object>(_modList[0]);
+                self->widgets[widx::checkbox_1].type = widget_type::checkbox;
+                self->widgets[widx::checkbox_1].text = roadExtraObject->name;
+
+                if (_lastSelectedMods & 1 << 0)
+                    activatedWidgets |= 1 << widx::checkbox_1;
+            }
+
+            if (_modList[1] != 0xFF)
+            {
+                auto roadExtraObject = objectmgr::get<road_extra_object>(_modList[1]);
+                self->widgets[widx::checkbox_2].type = widget_type::checkbox;
+                self->widgets[widx::checkbox_2].text = roadExtraObject->name;
+
+                if (_lastSelectedMods & 1 << 1)
+                    activatedWidgets |= 1 << widx::checkbox_2;
+            }
+        }
+        else
+        {
+            auto trackObj = objectmgr::get<track_object>(_trackType);
+
+            auto args = FormatArguments();
+            args.push(trackObj->name);
+
+            if (_modList[0] != 0xFF)
```

Comment:
Turn all of this into a loop over the 4 modlists. You could probably refactor the road one to loop over the 2 modlists as well.

```
for ( 1 : 4 )
{
    auto extraName = objectmgr...(modlist[i])->name;
    setCheckbox(i, extraName);
}
```